sdk/src/main/java/io/radar/sdk/RadarJobScheduler.kt

```diff
@@ -40,21 +42,58 @@ class RadarJobScheduler : JobService() {
                 putString(EXTRA_SOURCE, source.name)
             }
 
-            val jobInfo = JobInfo.Builder(JOB_ID, componentName)
+            val settings = RadarSettings.getFeatureSettings(context)
+            val newMaxConcurrentJobs = settings.maxConcurrentJobs
+            val currentMax = maxConcurrentJobs
```

Comment:
this whole section of 47-76 feels super complicated, there's probably a simpler way.  Maybe the values in the map should be AtomicInteger?  feels like `map.getOrDefault(0)` should be our friend when something is null vs needing complex initialization code that's hard to reason about.

Reply:
@tjulien I moved this logic to a separate class and added unit tests. Please take a look.